### PR TITLE
Utilities+Base: Improve `cal` and its man page

### PR DESCRIPTION
**Pull request by ronak69**
Full commit-messages:

```
Utilities: Improve `cal`

* Pad years with leading zeros.
* Error when `year` is specified with the option `--three-month-view`.
* Add a FIXME for handling cases where zeros are passed to arguments.
* Change wording of the option's descriptions to match current behaviour.

Base: Update `cal` man page

* New "Arguments" section.
* New "Errors and Return values" section to show erroneous input and the
  resultant error messages.
* New "Note on Calenders" subsection to specify that `cal` only supports
  the "Gregorian" calender.
* Add an example of using the option `--starting-day`.
* Better wording in long option descriptions that matches the current
  behaviour more accurately.
```

Initially i just wanted to fix a typo in the error message but i ended up making
all these changes. :)

Before `cal -3 2020` would not produce an error. After rearranging the code it now does.

`--three-month-view` and `--year` are not the best names for what they do. Three month view has nothing to do with the year being displayed in a 4x**3** grid. And `--year` is just a short hand for the current year and not the option that takes a year as an argument.

Some faults i found in other areas while working on this:
* markdown render doesn't preserve newlines in code blocks. (workaround is to use multiple code blocks but not used as this will get fixed eventually)
* the `ArgsParser` is treating "negative numbers" as "short options". (cal shouldn't allow negatives in arguments)

### CODE REVIEW

**Comment by ronak69** [comment]
I had an idea to do something special when displaying "October 10, 2018." But that would interfere with scriptability. Maybe Calender the application is the right place for that. :^)

File: Userland/Utilities/cal.cpp [hunk 1]
```diff
@@ -201,26 +191,37 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
-        for (int month_index = 1; month_index < 12; ++month_index) {
+        for (int month_index = 1; month_index <= 12;) {
             outln();
             outln();
             Vector<String> lines_left = TRY(month_lines_to_print(Header::Month, week_start_day, month_index++, year));
             Vector<String> lines_center = TRY(month_lines_to_print(Header::Month, week_start_day, month_index++, year));
-            Vector<String> lines_right = TRY(month_lines_to_print(Header::Month, week_start_day, month_index, year));
+            Vector<String> lines_right = TRY(month_lines_to_print(Header::Month, week_start_day, month_index++, year));
             print_months_side_by_side(lines_left, lines_center, lines_right);
         }
```
Comment (ronak69) [inline Userland/Utilities/cal.cpp:226]: I think this is more readable.

File: Userland/Utilities/cal.cpp [hunk 2]
```diff
@@ -201,26 +191,37 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     if (!month && year)
         year_mode = true;
 
+    if (three_month_mode && year_mode) {
+        warnln("cal: Cannot use --three-month-view when displaying a year.");
+        return 1;
```
Comment (ronak69) [inline Userland/Utilities/cal.cpp:196]: Is terminating the program the right choice? how about just a little warning? (`year_mode` will override `three_month_mode`)

Comment (AtkinsSJ) [reply Userland/Utilities/cal.cpp:196]: I'd say so. It doesn't make sense to request both, so terminating is fine.

**Pull request closed without merging**